docs/domain.md

```diff
@@ -0,0 +1,115 @@
+# Domains
+
+The convention used by this document is to explain paths in context of a
+com.endlessm.example.en app, and with files starting from the bundle
+root. So the file EKN_VERSION would be listed as
+`/share/ekn/data/com.endlessm.example.en/EKN_VERSION`.
+
+## Content ##
+
+A set of content is called a "domain". The format of content shipped in
+a domain is specified with a file,
+`/share/ekn/data/com.endlessm.example.en/EKN_VERSION`, shipped in each
+bundle, which contains a number. The currently supported version,
+described in this document, is 3.
+
+Domains have two major parts: 1. a Xapian database of records which is
+used for finding records by their tags for category searches, or doing
+queries based on article title or similar, and 2. one or more "shard
+files" which contain actual content. The "shard" is a custom archive or
+packfile format similar to .zip or .tar, but with Endless-specific
+enhacements.
+
+## Database ##
+
+Each record is accompanied by JSON metadata describing its title, its
+license, its tags, and other various kinds, along with content for it.
+This metadata is placed in the shard, indexed by its "EKN ID", a 20-byte
+identifier.
+
+The motivation for this setup was to cut down on the use of Xapian for
+critical path data and lookup, since each use of Xapian requires an HTTP
+request to a separate server.
+
+The Xapian database is embedded directly in the shard, allowing shard
+files to be completely self-contained pieces of searchable data. It
+is a Xapian single-file "Glass" database, containing a position table, a
+postlist table, a record table, etc.
+
+As an example of a metadata file, here's one:
+
+```
+    {
+      "@type": "ekn://_vocab/ImageObject",
+      "@id": "ekn:///96e17232727d9e86490160be01fbe45650d3ed90",
+      "copyrightHolder": "LoKiLeCh",
+      "caption": "Barbarossa chandelier",
+      "contentType": "image/jpeg",
+      "title": "File:Barbarossaleuchter.jpg",
+      "tags": [
+        "EknMediaObject"
+      ],
+      "license": "CC BY-SA 3.0",
+      "sourceURI": "https://upload.wikimedia.org/wikipedia/commons/2/2a/Barbarossaleuchter.jpg",
+      "lastModifiedDate": "2007-05-20T21:17:48",
+      "matchingLinks": [
+          "https://commons.wikimedia.org/wiki/File:Barbarossaleuchter.jpg",
+          "https://upload.wikimedia.org/wikipedia/commons/thumb/2/2a/Barbarossaleuchter.jpg/220px-Barbarossaleuchter.jpg"
+      ],
+      "width": 757
+      "height": 600,
+    }
+```
+
+Fields like "@type" and "@id" are part of [JSON-LD](http://json-ld.org/),
+a specification for hypermedia-style linked data.
```

Comment:
You should probably mention that we don't use "real" JSON-LD, and that these schemas are roughly based on schema.org equivalents.